lib/driver.js

```diff
@@ -295,26 +283,28 @@ class XCUITestDriver extends BaseDriver {
       await iosSettings.setPreferences(this.opts.device, this.opts, this.isSafari());
 
       let installAppPromise = null;
-      if (this.opts.app) {
-        if (await simBooted(this.opts.device)) {
-          installAppPromise = this.installApp();
-        } else {
-          installAppPromise = new B(async (resolve, reject) => {
-            this.opts.device.on(BOOT_COMPLETED_EVENT, async () => {
-              try {
-                await this.installApp();
-                resolve();
-              } catch (err) {
-                reject(err);
-              }
-            });
+      if (this.opts.app && !await this.opts.device.isRunning()) {
+        // Install app asynchronously as soon as device booting is completed
+        installAppPromise = new B(async (resolve, reject) => {
+          this.opts.device.on(BOOT_COMPLETED_EVENT, async () => {
+            try {
+              await this.installApp();
+              this.logEvent('appInstalled');
+              resolve();
+            } catch (err) {
+              reject(err);
+            }
           });
-        }
+        });
       }
       await this.startSim();
       this.logEvent('simStarted');
-      await installAppPromise;
-      this.logEvent('appInstalled');
+      if (installAppPromise) {
+        await installAppPromise;
+      } else if (this.opts.app) {
+        await this.installApp();
+        this.logEvent('appInstalled');
```

Comment:
We log this event only if installation of a custom app succeeds and don't do it if the app is already installed or a system application tested. Do you want this event to be logged for such cases as well?